datadog_lambda/cold_start.py

```diff
@@ -6,17 +6,28 @@
 logger = logging.getLogger(__name__)
 
 _cold_start = True
+_proactive_initialization = False
 _lambda_container_initialized = False
 
 
-def set_cold_start():
+def set_cold_start(init_timestamp_ns):
     """Set the value of the cold start global
 
     This should be executed once per Lambda execution before the execution
     """
     global _cold_start
     global _lambda_container_initialized
-    _cold_start = not _lambda_container_initialized
+    global _proactive_initialization
+    if not _lambda_container_initialized:
+        now = time.time_ns()
+        if (now - init_timestamp_ns) // 1_000_000_000 > 10:
+            _cold_start = False
+            _proactive_initialization = True
+        else:
+            _cold_start = not _lambda_container_initialized
```

Comment:
@astuyve I think on this case `_proactive_initialization = False` see my case from APM stats:

![Screenshot 2023-07-07 at 14 28 28](https://github.com/DataDog/datadog-lambda-python/assets/471109/c6b42684-4f99-48ad-a689-184eb32840af)

Reply:
We initialize `_proactive_initialization` to `False` on line 9 so we shouldn't need to redeclare that here.

In this case rolling up to `top 10` is skewing the results:
<img width="410" alt="image" src="https://github.com/DataDog/datadog-lambda-python/assets/1598537/7c50f95a-6205-4f32-91ec-243c2fc18d88">
I can see something similar in one of my orgs:
<img width="1313" alt="image" src="https://github.com/DataDog/datadog-lambda-python/assets/1598537/8ca17a7d-5074-41b3-a0a1-2c08744c4960">
But when I remove the `pie chart` grouping, I see many `cold_start: true` spans:
<img width="1320" alt="image" src="https://github.com/DataDog/datadog-lambda-python/assets/1598537/1f34fa87-2b49-4dff-854d-f5bb057cdd59">

However for clarity, line 27 should always be `false` as we're already inside of a conditional checking for `_lambda_container_initialized == false`